static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-epic-liveblog.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+import { makeABTest } from 'common/modules/commercial/contributions-utilities';
+import { logView } from 'common/modules/commercial/acquisitions-view-log';
+import template from 'lodash/utilities/template';
+import $ from 'lib/$';
+import config from 'lib/config';
+import mediator from 'lib/mediator';
+import ElementInView from 'lib/element-inview';
+import fastdom from 'lib/fastdom-promise';
+import liveblogEpicTemplate from 'raw-loader!common/views/acquisitions-epic-liveblog.html';
+import { liveblog as liveblogCopy } from 'common/modules/commercial/acquisitions-copy';
+
+const pageId = config.page.pageId || '';
+
+let isAutoUpdateHandlerBound = false;
+
+const INSERT_EPIC_AFTER_CLASS = 'js-insert-epic-after';
+
+const getLiveblogEntryTimeData = el => {
+    const $timeEl = $('time', el);
+
+    return {
+        datetime: $timeEl.attr('datetime'),
+        title: $timeEl.attr('title'),
+        date: $timeEl.text(),
+        time: $('.block-time__absolute', el).text(),
+    };
+};
+
+const setEpicLiveblogEntryTimeData = (el, timeData) => {
+    if (!el) {
+        return;
+    }
+
+    const $epicTimeEl = $('time', el);
+    $epicTimeEl.attr('datetime', timeData.datetime);
+    $epicTimeEl.attr('title', timeData.title);
+    $epicTimeEl.text(timeData.date);
+    $('.block-time__absolute', el).text(timeData.time);
+};
+
+const setupViewTracking = (el, test) => {
+    // top offset of 18 ensures view only counts when half of element is on screen
+    const elementInView = ElementInView(el, window, {
+        top: 18,
+    });
+
+    elementInView.on('firstview', () => {
+        logView(test.id);
+        mediator.emit(test.viewEvent);
+    });
+};
+
+const addEpicToBlocks = (epicHtml, test) =>
+    fastdom.write(() => {
+        const $blocksToInsertEpicAfter = $(`.${INSERT_EPIC_AFTER_CLASS}`);
```

Comment:
This should be wrapped into `fastdom.read` instead. If you don't want to refactor it now, it would be nice to leave a comment :)